deployer/scripts/install.sh

```diff
@@ -255,7 +255,13 @@ function generate_configmaps() {
     # generate curator configmap
     oc create configmap logging-curator \
       --from-file=config.yaml=conf/curator.yml
-    oc label configmap/logging-curator logging-infra=support # make easier to delete later
+    oc label configmap/logging-curator logging-infra=support
+
+    # generate fluentd configmap
+    oc create configmap logging-fluentd \
+      --from-file=fluent.conf=conf/fluent.conf \
+      --from-file=throttle-config.yaml=conf/fluentd-throttle-config.yaml || return
```

Comment:
I agree it's not an easy setup to explain or understand.

Having all the files in the configmap was my first design. But when we
started talking about upgrades and users modifying things I realized I
wanted as little in their hands as I could get away with.

It does make some sense to have the dynamically generated stuff all in the
configmap, but then it would have to be generated in the deployer, not in
fluentd, and the throttle conf generator is in ruby to match fluentd... so
that seemed ugly.

On Wed, Jul 13, 2016 at 1:05 PM, Anton <notifications@github.com> wrote:

> In deployer/scripts/install.sh
> <https://github.com/openshift/origin-aggregated-logging/pull/195#discussion_r70666276>
> :
>
> > @@ -255,7 +255,13 @@ function generate_configmaps() {
> >      # generate curator configmap
> >      oc create configmap logging-curator \
> >        --from-file=config.yaml=conf/curator.yml
> > -    oc label configmap/logging-curator logging-infra=support # make easier to delete later
> > +    oc label configmap/logging-curator logging-infra=support
> > +
> > +    # generate fluentd configmap
> > +    oc create configmap logging-fluentd \
> > +      --from-file=fluent.conf=conf/fluent.conf \
> > +      --from-file=throttle-config.yaml=conf/fluentd-throttle-config.yaml || return
>
> My primary concern is that right now there are 4 different places where
> fluentd configuration is defined:
> 1) hardcoded config files
> 2) dynamically generated config files
> 3) config files passed via configmaps
> 4) env variables
>
> I think this is not an easy setup.
> User needs to have knowledge of fluentd, then user also needs to
> understand 1)-4) to know where to add his modifications(because w/o knowing
> the full config and where it is coming from it is very easy to screw up).
> On update there are 4 different items to track.
>
> To me personally it is much easier to see the entire fluentd configuration
> accurately(i.e. dynamically generated configs output to configmap, rather
> than to .conf file) taken from configmaps.
> I'm not arguing against good directory structure which will advise
> how/where users need to add their modifications.
> If everything(most stuff) is in configmaps on upgrade you can still check
> what was edited and act accordingly.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin-aggregated-logging/pull/195/files/a46c725bc59c9ac2d21d547550c583493e5b9ce7#r70666276>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABz-tm7mUtQZvlF8ZUK5xV6cpFJxp-pks5qVRrpgaJpZM4JHWV8>
> .
>